src/control/server/mgmt_client.go

```diff
@@ -41,37 +41,75 @@ const (
 )
 
 type (
-	mgmtSvcClientCfg struct {
+	// GrpcClientCfgHelper provides config access.
+	GrpcClientCfgHelper interface {
+		GetConfig() *grpcClientCfg
+		SetConfig(grpcClientCfg)
+		LeaderAddress() (string, error)
+	}
+	// grpcClientCfgHelper implements GrpcClientCfgHelper.
+	grpcClientCfgHelper struct {
+		cfg grpcClientCfg
+	}
+	// GrpcClient provides a subset of MgmtSvcClient for operations between gRPC servers.
+	GrpcClient interface {
+		GrpcClientCfgHelper
+		Join(context.Context, *mgmtpb.JoinReq) (*mgmtpb.JoinResp, error)
+		PrepShutdown(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Stop(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Start(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Query(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+	}
+	// grpcClient implements GrpcClient.
+	grpcClient struct {
+		log logging.Logger
+		*grpcClientCfgHelper
+	}
+	grpcClientCfg struct {
 		AccessPoints    []string
 		ControlAddr     *net.TCPAddr
 		TransportConfig *security.TransportConfig
 	}
-	mgmtSvcClient struct {
-		log logging.Logger
-		cfg mgmtSvcClientCfg
-	}
 )
 
-func newMgmtSvcClient(ctx context.Context, log logging.Logger, cfg mgmtSvcClientCfg) *mgmtSvcClient {
-	return &mgmtSvcClient{
-		log: log,
-		cfg: cfg,
+func (gcch *grpcClientCfgHelper) GetConfig() *grpcClientCfg {
+	return &gcch.cfg
+}
+
+func (gcch *grpcClientCfgHelper) SetConfig(cfg grpcClientCfg) {
+	gcch.cfg = cfg
+}
+
+func (gcch *grpcClientCfgHelper) LeaderAddress() (string, error) {
+	if len(gcch.cfg.AccessPoints) == 0 {
+		return "", errors.New("no access points defined")
+	}
+
+	// TODO: Develop algorithm for determining current leader.
+	// For now, just choose the first address.
+	return gcch.cfg.AccessPoints[0], nil
+}
+
+func newGrpcClient(ctx context.Context, log logging.Logger, cfg grpcClientCfg) GrpcClient {
+	return &grpcClient{
+		log:                 log,
+		grpcClientCfgHelper: &grpcClientCfgHelper{cfg: cfg},
 	}
 }
 
 // delayRetry delays next retry.
-func (msc *mgmtSvcClient) delayRetry(ctx context.Context) {
+func (gc *grpcClient) delayRetry(ctx context.Context) {
 	select {
 	case <-ctx.Done(): // break early if the parent context is canceled
 	case <-time.After(retryDelay): // otherwise, block until after the delay duration
 	}
 }
 
-func (msc *mgmtSvcClient) withConnection(ctx context.Context, ap string,
+func (gc *grpcClient) withConnection(ctx context.Context, ap string,
 	fn func(context.Context, mgmtpb.MgmtSvcClient) error) error {
```

Comment:
If you change the signature of this closure to accept the smaller interface defined above, it should be possible to test all of this code using the mock client that you've added along with the new interface.